### PR TITLE
Finalise 25 test case for sprint 2

### PR DESCRIPTION
**Pull request by Kang-Quan**
Added testcases to these files/controller
policy
contacts
category product
product details
pages/search
context/search
searchinput
getProductController
getSingleProductController
productPhotoController
productFiltersController
productCountController
productListController
searchProductController
realtedProductController
productCategoryController

### CODE REVIEW

File: client/src/components/Form/SearchInput.test.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		expect(screen.getByPlaceholderText(/search/i)).toBeInTheDocument();
+		// Check if the search button is rendered
+		expect(
+			screen.getByRole("button", { name: /search/i })
+		).toBeInTheDocument();
+
+		const searchInput = screen.getByPlaceholderText(/Search/i);
+		const searchButton = screen.getByRole("button", { name: /search/i });
+
+        fireEvent.change(searchInput, { target: { value: "test keyword" } });
```
Comment (eugenetangkj) [inline client/src/components/Form/SearchInput.test.js:57]: I think the indentation is a bit off this part haha

File: client/src/pages/ProductDetails.test.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        expect(await screen.findByText(/Similar Product 1/i)).toBeInTheDocument();
+		expect(await screen.findByText(/\$90.00/i)).toBeInTheDocument();
+		expect(await screen.findByText(/Similar product description 1..../i)).toBeInTheDocument();
+
+        
+        expect(await screen.findByText(/Similar Product 2/i)).toBeInTheDocument();
+		expect(await screen.findByText(/\$80.00/i)).toBeInTheDocument();
+		expect(await screen.findByText(/Similar product description 2..../i)).toBeInTheDocument();
+        
+        const moreDetailsButtons = await screen.findAllByText(/More Details/i);
```
Comment (eugenetangkj) [inline client/src/pages/ProductDetails.test.js:134]: Just my opinion, but this test seems to be doing more than 1 thing. You are trying to determine whether product details are rendered correctly with non-empty similar products. However, after checking for similar products, you proceed with simulating pressing of more details button, which seems to no longer bepart of the test case objective anymore.

**Review by eugenetangkj — APPROVED** [review]
Looks great, thanks!

File: client/src/pages/ProductDetails.test.js (hunk 2 above)
Comment (Kang-Quan) [reply client/src/pages/ProductDetails.test.js:134]: I agree with you, if i have more testcases I would have split them up for more clarity, in this test I was checking for all successful actions where i pass in all valid inputs and use the expect function to ensure the code is running expectedly including button rendering and its functionality.

**Pull request merged**